schema/rdl_schema_0.1.json

```diff
@@ -1710,13 +1552,502 @@
+          "description": "The type of analysis applied to the hazard data used in the modelled scenario(s), from the closed [analysis type codelist]https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists.html#analysis_type).",
+          "codelist": "analysis_type.csv",
+          "openCodelist": false,
+          "enum": [
+            "deterministic",
+            "empirical",
+            "probabilistic"
+          ]
+        },
+        "intensity": {
+          "title": "Hazard intensity measurement",
+          "description": "The metric and units the hazard intensity measurement is given in, from the open [IMT codelist]https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists.html#IMT).",
+          "type": "string",
+          "codelist": "IMT.csv",
+          "openCodelist": true,
+          "minLength": 1
+        },
+        "category": {
+          "title": "Exposure category",
+          "description": "The category of the exposed assets, from the closed [exposure category codelist](https://rdl-standard.readthedocs.io/en/{{version}}/data_model/codelists/#exposure_category).",
+          "type": "string",
+          "codelist": "exposure_category.csv",
+          "openCodelist": false,
+          "enum": [
+            "Agriculture",
+            "Buildings",
+            "Infrastructures",
+            "Population",
+            "Natural environment"
+          ]
+        },
+        "cost": {
+          "title": "Asset cost",
+          "type": "array",
+          "description": "The exposure costs associated with specific elements of assets detailed in the dataset.",
+          "items": {
+            "$ref": "#/$defs/Cost"
+          },
+          "minItems": 1,
+          "uniqueItems": true
+        },
+        "taxonomy": {
+          "title": "Exposure taxonomy scheme",
+          "type": "string",
+          "description": "The name of the taxonomy scheme used to create descriptive individual asset feature strings within the dataset. Use of GED4ALL is recommended.",
+          "minLength": 1
+        },
+        "impact": {
+          "title": "Vulnerability impact",
+          "type": "object",
+          "description": "The details of the vulnerability impact values produced in the modelled scenario(s).",
+          "$ref": "#/$defs/Impact"
+        },
+        "spatial": {
+          "title": "Vulnerability spatial details",
+          "description": "Spatial details of the dataset. `.scale` must be provided, other properties of `Location` can be included if they are different from those already detailed in the top level `spatial` object.",
+          "$ref": "#/$defs/Location",
+          "required": [
+            "scale"
+          ]
+        },
+        "functions": {
+          "title": "Vulnerability functions",
+          "type": "object",
+          "description": "Details of the functions used to calculate the vulnerability dataset.",
+          "properties": {
```

Comment:
Given that the `approach` and `relationship` properties are common to all function types, they should go here to reduce repetition. The properties that are specific to particular function types can remain under the key for the relevant function type (e.g. `fragility`, `damage_to_loss` etc.) and the descriptions of those keys can be updated to 'Details specific to <function type> functions'.

Reply:
I had initially done that then talked myself out of it! I'll move them back to their own definitions to be referenced

Reply:
they are repeated but a vulnerability dataset can involve more than one function type and the approach and relationship might be different for each function type which is why they need to be kept within each function type.

to reduce repetition though I'll move the common fields into their own definitions.

Reply:
And now I remember why I talked myself out of it, for the same reason `category` isn't a seperate definition even though it appears in multiple places. There'll just need to be a bunch of repetition I think.

Reply:
Makes sense, thanks!